### PR TITLE
Multitool data disk wiping

### PR DESCRIPTION
**Pull request by penny64**
[FEATURE] [QOL]
## About the PR 

image image
image image

image

The multitool was able to wipe the memory core on the mainframe, I've

- updated it to modern grammar standards
- replaced waiting with actionbars
- added a zap effect identical to the multitool's special
- extended it to support all data disks- floppies, hard drives, tapes etc.
- attempting to wipe a read only disk backfires on the user

As an effect, since disks are wiped with the zap effect, holding the disk while wiping it zaps the user- as expected.

Based on input I've made two more changes

- Security clone disks are now read only
- The cloner will ignore the read only switch for the loading/deletion process only

## Why's this needed? 

This old crusty feature needed some love, and might needed to apply to more devices to actually fill a niche. Only applying to the rarest storage device left this even more obscure than it will be now.
We stay in scope by mostly only existing to wipe user data, as with the mainframe memory core.

## Considerations
 This is mostly silly- most disks are read only, leaving you with a few things like disks users actually used to store data. The main purpose of this is to make it more obvious that this functionality exists at all.

Disks that are read only include
- COMmaster
- DWAINE backup and master tapes
- Every disk in SoftTech except OS disks
- Security Clone Disks

None of these can be wiped.
The cloner will now load/delete clone records from read only disks, slightly inconsistent but speaking about their real world counterparts, floppy read only switches are handled in software- any drive could choose to ignore it.

## Changelog

```changelog
(u)Penny
(+)Multitools can now wipe storage devices
(+)Security disks are now read only, and read only disks can be read in the cloner
```

### CODE REVIEW

**Comment by nefarious6th** [comment]
My sticking point is the data disks. I don't care for this at all with that included.

**Comment by penny64** [comment]
> My sticking point is the data disks. I don't care for this at all with that included.

What issues do you forsee with wiping data disks- the auth disk and most software disks are read only, for example.
Clone disks are the clearest thing are you worried about those getting regularly wiped?

**Comment by nefarious6th** [comment]
Yes, the cloning disks are the sticking point for me. I'm not interested in lawyering it out. I'm generally not interested in anything that makes a frustrating and difficult job in the game even more frustrating or paranoid. That's all.

**Comment by penny64** [comment]
> Yes, the cloning disks are the sticking point for me. I'm not interested in lawyering it out. I'm generally not interested in anything that makes a frustrating and difficult job in the game even more frustrating or paranoid. That's all.

And whhat if they were read only?

**Comment by nefarious6th** [comment]
Probably the easiest solution, actually. Good thinking! I wasn't aware that that read-only on the cloner wasn't relevant.

**Comment by Carbadox** [comment]
I really dont like the idea of someone walking into tech storage and wiping all the ComMaster disks after destroying all the comm dishes

**Comment by penny64** [comment]
> I really dont like the idea of someone walking into tech storage and wiping all the ComMaster disks after destroying all the comm dishes

Are those read write?

File: code/obj/machinery/computer/cloning.dm [hunk 1]
```diff
@@ -764,11 +764,12 @@ proc/find_ghost_by_key(var/find_key)
 				src.diskette = null
 				. = TRUE
 		if("load")
-			if (src.diskette.read_only)
+			// Read only switches on floppy disks are software, the cloner can ignore it for loading
+			/*if (src.diskette.read_only)
```
Comment (ZeWaka) [inline code/obj/machinery/computer/cloning.dm:768]: why comment this out instead of remove it

File: code/modules/networks/computer3/disks.dm [hunk 2]
```diff
@@ -65,6 +65,26 @@
 
 		return D
 
+	proc/wipe_or_zap(mob/user)
+		if(!read_only)
+			user.visible_message("<span class='alert'><b>[user] wipes the [src.name]!</b></span>")
+			//qdel(src.root)
```
Comment (ZeWaka) [inline code/modules/networks/computer3/disks.dm:71]: bap

File: tgui/packages/tgui/interfaces/CloningConsole.js [hunk 3]
```diff
@@ -243,10 +242,10 @@ const Functions = (props, context) => {
           <Box>
             <Icon
               color={diskReadOnly ? 'bad' : 'good'}
-              name={diskReadOnly ? 'times' : 'check'}
+              name={'check'}
             />
             {' '}
-            {diskReadOnly ? 'Disk is read only.' : 'Disk is writeable.'}
```
Comment (ZeWaka) [inline tgui/packages/tgui/interfaces/CloningConsole.js:249]: what's with the wording change? i thought it was clear previously

Comment (penny64) [reply tgui/packages/tgui/interfaces/CloningConsole.js:249]: I had crossed wires that "Disk is read only." meant "disk is unloadable", when that was in fact, what I was changing

**Pull request merged**